src/App.js

```diff
@@ -1,5 +1,146 @@
+			let number = Random.pickNumberInRange(1, 9);
+			if (!this.computerNumArr.includes(number))
+				this.computerNumArr.push(number);
+		}
+
+		return this.computerNumArr;
+	}
+
+	playGame() {
+		Console.readLine(
+			MESSAGES.REQUIRE_USER_INPUT(NUMBERS.GAME_MAX),
+			userAnswerStr => {
+				this.isValidNumber(userAnswerStr);
+
+				const [ball, strike] = this.compareComputerWithUser(
+					this.computerNumArr,
+					userAnswerStr
+				);
+				const resultMessage = this.convertScoreToMessage([
+					ball,
+					strike,
+				]);
+
+				this.showResultMessage(resultMessage);
+			}
+		);
+	}
+
+	isValidNumber(userAnswerStr) {
+		const answer = userAnswerStr.replace(REGEX.SPACE, '');
+		const { length } = answer;
+
+		if (length >= 4) throw new Error(MESSAGES.INVALID_LENGTH);
+
+		if (isNaN(answer)) throw new Error(MESSAGES.NOT_A_NUMBER);
+
+		if (this.removeDuplicated(userAnswerStr) !== userAnswerStr)
+			throw new Error(MESSAGES.DUPLICATED_NUM);
+
+		return true;
+	}
+
+	removeDuplicated(s) {
+		let answer = '';
+
+		for (let i = 0; i < s.length; i++) {
+			if (s.indexOf(s[i]) === i) answer += s[i];
+		}
+
+		return answer;
+	}
+
+	isValidChoice(playerChoice) {
+		if (!REGEX.CHOICE.test(playerChoice)) {
+			Console.print(MESSAGES.FORMAT_ERROR_CHOICE);
+			throw Error(MESSAGES.FORMAT_ERROR_CHOICE);
+		}
+		return true;
+	}
+
+	compareComputerWithUser(computerNumArr, userAnswerStr) {
+		let [ball, strike] = [0, 0];
+
+		userAnswerStr.split('').forEach((userAnswerStr, userAnswerIdx) => {
+			this.isValidNumber(userAnswerStr);
+
+			computerNumArr.map((computerNum, computerNumIdx) => {
+				if (
+					computerNum === Number(userAnswerStr) &&
+					computerNumIdx === userAnswerIdx
+				) {
+					strike++;
+					return;
+				}
+				if (computerNum === Number(userAnswerStr)) ball++;
+			});
+		});
+
+		return [ball, strike];
+	}
+
+	convertScoreToMessage([ball, strike]) {
+		let resultMessage = '';
+
+		if (strike > 0 && ball > 0) {
+			resultMessage = `${ball}${UNITS.BALL} ${strike}${UNITS.STRIKE}`;
+		} else if (strike > 0) {
+			resultMessage = `${strike}${UNITS.STRIKE}`;
+		} else if (ball > 0) {
+			resultMessage = `${ball}${UNITS.BALL}`;
+		} else {
+			resultMessage = UNITS.NOTHING;
+		}
+
+		return resultMessage;
+	}
+
+	showResultMessage(resultMessage) {
+		if (resultMessage === MESSAGES.THREE_STRIKE) {
+			Console.print(MESSAGES.THREE_STRIKE);
+			Console.print(MESSAGES.END(3));
+			this.askToReplay();
+		} else {
+			Console.print(resultMessage);
+			this.playGame();
+		}
+	}
+
+	askToReplay() {
+		Console.readLine(MESSAGES.ASK_RESTART, playerChoice => {
+			this.isValidChoice(playerChoice);
+
+			if (playerChoice === CHOICE.PLAY_AGAIN) {
+				this.initGame();
+				this.playGame();
+			} else if (playerChoice === CHOICE.EXIT) {
+				Console.print(MESSAGES.END_GAME);
+				Console.close();
+			} else {
+				Console.print(MESSAGES.FORMAT_ERROR_CHOICE);
+				throw Error(MESSAGES.FORMAT_ERROR_CHOICE);
+			}
```

Comment:
```suggestion
			this.isValidChoice(playerChoice);

			(playerChoice === CHOICE.PLAY_AGAIN) ? this.restartGame() : this.endGame();
```
isValidChoice 메서드에서 예외처리 테스트를 충분히 많이 하셨다면, else문의 에러처리는 하지 않으셔도 될 것이라 생각되어 restartGame()과 endGame()메서드 추가 및 삼항연산자 활용으로 가독성을 챙기시는건 어떨까요??